vnext/ReactWindowsCore/WinRTWebSocketResource.h

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// clang-format off
+
+#pragma once
+
+#include <winrt/Windows.Networking.Sockets.h>
+#include <winrt/Windows.Storage.Streams.h>
+#include <IWebSocketResource.h>
+
+#include <future>
+#include <queue>
+
+namespace Microsoft::React
+{
+
+class WinRTWebSocketResource : public IWebSocketResource
+{
+  winrt::Windows::Foundation::Uri m_uri;
+  winrt::Windows::Networking::Sockets::MessageWebSocket m_socket;
+  winrt::Windows::Networking::Sockets::MessageWebSocket::MessageReceived_revoker m_revoker;
+  winrt::Windows::Storage::Streams::DataWriter m_writer{ m_socket.OutputStream() };
+
+  std::atomic_bool m_connectRequested;
+  std::atomic_bool m_sendRequested;
+  std::promise<void> m_connectPerformed;
+  std::promise<void> m_closePerformed;
+  winrt::Windows::Foundation::IAsyncAction m_connectAction;
+
+  CloseCode m_closeCode{ CloseCode::None };
+  std::string m_closeReason;
+  std::queue<std::pair<std::string, bool>> m_writeQueue;
```

Comment:
>std::string [](start = 23, length = 11)

I think std::string is not right here because you're using uint8_t elsewhere?

Reply:
This container is ultimately filled by calling `IWebSocketResource::Send` and `...SendBinary`, which expect `std::string`.

Plus, the "binary" write uses WinRT's `IBuffer`, which doesn't deal with `uint8_t`.